### PR TITLE
Added debug display for rays/frame

### PR DESCRIPTION
**Pull request by nerites**
### Purpose of this PR
image

This PR doesn't have rays/second support yet- this will be added in a separate branch on top of this changeset. 

---
### Release Notes
Sourced heavily from Ionut's hackweek project. Major point of deviation is using a shader to draw results to the screen, instead of doing an AsyncReadback to write to a text GameObject. 

Rebased https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/2811 on HDRP/ReflectionFilter.

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=hdrtr_reflectionfilter_countrays&unity_branch=graphics%2Fraytracing%2Fdxr&automation-tools_branch=add-platform-filter

**Manual Tests**: What did you do?
Verified overlay looked correct and updated as I panned around the screen in the scene view. 
Confirmed that issue with missing area shadows occurred on HDRP/ReflectionFilter without my changes.

**Automated Tests**: What did you setup?
Nothing

Any test projects to go with this to help reviewers?
None

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?
Low- most of the trickiness was in getting texture formats to play nice. 

**Halo Effect**: None, Low, Medium, High?
Low- the shader code that was added does not change existing shader code, and the invocation of the raycount debug is very self-contained.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDRaytracingAmbientOcclusion.cs [hunk 1]
```diff
@@ -88,36 +88,41 @@ public void RenderAO(HDCamera hdCamera, CommandBuffer cmd, RTHandleSystem.RTHand
...
-            // Run the calculus
-            cmd.DispatchRays(aoShader, m_RayGenShaderName, (uint)hdCamera.actualWidth, (uint)hdCamera.actualHeight, 1);
+                cmd.SetRaytracingIntParam(aoShader, HDShaderIDs._RayCountEnabled, m_RaytracingManager.rayCountManager.rayCountEnabled);
+                cmd.SetRaytracingTextureParam(aoShader, m_RayGenShaderName, HDShaderIDs._RayCountTexture, m_RaytracingManager.rayCountManager.rayCountTex);
 
-            using (new ProfilingSample(cmd, "Filter Reflection", CustomSamplerId.RaytracingAmbientOcclusion.GetSampler()))
+                // Run the calculus
+                cmd.DispatchRays(aoShader, m_RayGenShaderName, (uint)hdCamera.actualWidth, (uint)hdCamera.actualHeight, 1);
+            }
+            using (new ProfilingSample(cmd, "Filter AO", CustomSamplerId.RaytracingFilterAO.GetSampler()))
```
Comment (nerites) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDRaytracingAmbientOcclusion.cs]: These changes are just to bracket all this in a new ProfilingSample and add bindings for rayCountEnabled and rayCountTex.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDRaytracingReflection.cs [hunk 2]
```diff
@@ -85,105 +85,112 @@ public void RenderReflections(HDCamera hdCamera, CommandBuffer cmd, RTHandleSyst
             ComputeShader bilateralFilter = m_PipelineAsset.renderPipelineResources.shaders.reflectionBilateralFilterCS;
             RaytracingShader reflectionShader = m_PipelineAsset.renderPipelineResources.shaders.reflectionRaytracing;
             bool missingResources = rtEnvironement == null || blueNoise == null || bilateralFilter == null || reflectionShader == null;
+            using (new ProfilingSample(cmd, "Raytrace Reflections", CustomSamplerId.RaytracingIntegrateReflection.GetSampler()))
```
Comment (nerites) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDRaytracingReflection.cs]: These changes look bigger than they are- I just bracketed all this in a new ProfilingSample and then added bindings for rayCountEnabled and rayCountTex.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/Shaders/RaytracingAmbientOcclusion.raytrace [hunk 3]
```diff
@@ -16,33 +16,33 @@
 #include "Packages/com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/Shaders/RaytracingSampling.hlsl"
 
 // The target acceleration structure that we will evaluate the reflexion in
```
Comment (nerites) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/Shaders/RaytracingAmbientOcclusion.raytrace]: The changes to this file are:
- Standardizing whitespace from tabs -> 4 spaces
- Add write to rayCountTex

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipelineAsset.cs [hunk 4]
```diff
@@ -259,6 +259,9 @@ public void EvaluateSettings()
             if(needUpdate)
             {
                 UnityEditor.PlayerSettings.SetScriptingDefineSymbolsForGroup(UnityEditor.BuildTargetGroup.Standalone, string.Join(";", defineArray.ToArray()));
+#if ENABLE_RAYTRACING
+                m_RenderPipelineResources.LoadRayTraceShaders();
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipelineAsset.cs]: I get the point about this call. But it is not the right place to call it. We should make a correct call to init all missing resources if any is missing. Please remove this call for now and we will do that at another place in another PR.
